### PR TITLE
Add optional boxes behind callouts

### LINKED ISSUE #5: Frame-specific call-outs mid-animation
Ideally, would be able to reference frames by date rather than frame number

MVP: show just text 

Slow down frames and visually focus on certain gages over others (everything but the important region is greyed out). Would need to make it obvious that time has slowed. The "outro" part of viz storm gif has an example of this.

### PR DESCRIPTION
**Pull request by lindsaycarr**
You need to add `addBox: TRUE` with the callout in `viz_config.yml` in order to get one to appear. Fixes #5.

image

I ran the following code to generate the image above to verify that all callouts look as expected:

```
test_plot_funs(view_fun, basemap_fun, watermark_fun, legend_fun, 
               datetime_fun_20180915_00, gage_sites_fun_20180915_00, callouts_fun_20180715_00, 
               callouts_fun_20180221_00, callouts_fun_20171210_00, callouts_fun_20180914_00, 
               callouts_fun_20180922_00, png_file = 'test.png')
```

### CODE REVIEW

File: viz_config.yml [hunk 1]
```diff
@@ -89,6 +89,7 @@ callouts:
         y_loc: 0.6
         pos: 3
         cex: 2
+    addBox: TRUE
```
[the remark this answers is not in the document]
Comment (lindsaycarr) [reply viz_config.yml]: ack! yes...overlooked that :)

File: 6_visualize/src/prep_callouts_fun.R [hunk 2]
```diff
@@ -25,18 +25,63 @@ prep_callouts_fun <- function(callouts_cfg, dateTime){
...
-            y_i <- y - (i-1)*strheight(callouts[i])*2
+            y_i <- y - (i-1)*strheight(callouts[i])*font_y_multiplier
             text(x, y_i, labels = callouts[i],
                  cex = callout_text_cfg_n$cex,
                  pos = callout_text_cfg_n$pos,
                  col = 'grey40')
           }
+
+          if(addBox) {
+            max_strwidth <- max(strwidth(callouts))*font_x_multiplier # buffer for abel since it can't do that correctly
```
[the remark this answers is not in the document]
Comment (lindsaycarr) [reply 6_visualize/src/prep_callouts_fun.R]: it was throwing a warning about not knowing the "abel" font. Let me plug that in and see what happens

[the remark this answers is not in the document]
Comment (lindsaycarr) [reply 6_visualize/src/prep_callouts_fun.R]: Hmmm I can't seem to get that to work.

[the remark this answers is not in the document]
Comment (aappling-usgs) [reply 6_visualize/src/prep_callouts_fun.R]: Hmm, oh well. Glad you tried.

**Pull request merged**